src/main/java/gyro/google/compute/AbstractInstanceGroupManagerResource.java

```diff
@@ -0,0 +1,446 @@
+        if (autoHealingPolicy == null) {
+            autoHealingPolicy = new ArrayList<>();
+        }
+        return autoHealingPolicy;
+    }
+
+    public void setAutoHealingPolicy(List<ComputeInstanceGroupManagerAutoHealingPolicy> autoHealingPolicy) {
+        this.autoHealingPolicy = autoHealingPolicy;
+    }
+
+    /**
+     * An optional description of this resource. Provide this property when you create the resource.
+     */
+    public String getDescription() {
+        return description;
+    }
+
+    public void setDescription(String description) {
+        this.description = description;
+    }
+
+    /**
+     * Fingerprint of this resource. This field may be used in optimistic locking. It will be ignored when inserting an InstanceGroupManager. An up-to-date fingerprint must be provided in order to update the InstanceGroupManager, otherwise the request will fail with error 412 conditionNotMet.To see the latest fingerprint, make a get() request to retrieve an InstanceGroupManager.
+     */
+    public String getFingerprint() {
+        return fingerprint;
+    }
+
+    public void setFingerprint(String fingerprint) {
+        this.fingerprint = fingerprint;
+    }
+
+    /**
+     * The URL of the instance template that is specified for this managed instance group. The group uses this template to create all new instances in the managed instance group.
+     *
+     * @resource gyro.google.compute.InstanceTemplateResource
+     */
+    @ConflictsWith("version")
+    @Updatable
+    public InstanceTemplateResource getInstanceTemplate() {
+        return instanceTemplate;
+    }
+
+    public void setInstanceTemplate(InstanceTemplateResource instanceTemplate) {
+        this.instanceTemplate = instanceTemplate;
+    }
+
+    /**
+     * Named ports configured for the Instance Groups complementary to this Instance Group Manager.
+     *
+     * @subresource gyro.google.compute.ComputeNamedPort
+     */
+    public List<ComputeNamedPort> getNamedPort() {
+        if (namedPort == null) {
+            namedPort = new ArrayList<>();
+        }
+
+        return namedPort;
+    }
+
+    public void setNamedPort(List<ComputeNamedPort> namedPort) {
+        this.namedPort = namedPort;
+    }
+
+    // TODO: target pool resource
+    // https://github.com/perfectsense/gyro-google-provider/issues/79
+    //    /**
+    //     * The URLs for all TargetPool resources to which instances in the instanceGroup field are added. The target pools automatically apply to all of the instances in the managed instance group.
+    //     */
+    //    public List<AbstractTargetPoolResource> getTargetPools() {
+    //        if (targetPools == null) {
+    //            targetPools = new ArrayList<>();
+    //        }
+    //
+    //        return targetPools;
+    //    }
+    //
+    //    public void setTargetPools(List<AbstractTargetPoolResource> targetPools) {
+    //        this.targetPools = targetPools;
+    //    }
```

Comment:
Should be removed and be done as a separate PR when target pool is available

Reply:
Yeah, it should be handled in a separate PR, but better to leave the comment as it will be helpful when implementing it.